test/e2e/multikueue/e2e_test.go

```diff
@@ -1126,3 +1075,17 @@ func GetMultiKueueClusterNameFromAdmissionCheckMessage(message string) string {
 	}
 	return ""
 }
+
+func checkResourcesNotFoundEventually(ctx context.Context, clients []client.Client, wlLookupKey client.ObjectKey, objects []client.Object) {
```

Comment:
Can we reuse util.ExpectObjectToBeDeleted instead? TBH we don't need this function to support multiple objects, it can be confusing and is unusued this way.

I would just suggest to introduce `expectObjectToBeDeletedOnWorkerClusters`, and even avoid repeating `[]client.Client{k8sWorker1Client, k8sWorker2Client}` for now.

Longer term I think we could have a single function ` util.ExpectObjectToBeDeleted` with variadic number of clients.